static/js/drafts.js

```diff
@@ -419,23 +426,25 @@ exports.drafts_handle_events = function (e, event_key) {
     if (event_key === "backspace" || event_key === "delete") {
         if (elt.parentElement.hasAttribute("data-draft-id")) {
             var focus_draft_back_row = $(elt)[0].parentElement;
-            var backnext_focus_draft_row = $(focus_draft_back_row).next();
-            var backprev_focus_draft_row = $(focus_draft_back_row).prev();
+            var draft_row = $(focus_draft_back_row);
+            var backnext_focus_draft_row = draft_row.next();
+            var backprev_focus_draft_row = draft_row.prev();
             var delete_id;
+
+            // Try to get the next draft in the list and 'focus' it
+            // Use previous draft as a fallback
             if (backnext_focus_draft_row[0] !== undefined) {
                 delete_id = backnext_focus_draft_row[0].getAttribute("data-draft-id");
             } else if (backprev_focus_draft_row[0] !== undefined) {
                 delete_id = backprev_focus_draft_row[0].getAttribute("data-draft-id");
             }
-            drafts.draft_model.deleteDraft(focused_draft);
-            document.activeElement.parentElement.remove();
+
```

Comment:
This refactoring isn't completely obvious to me.  Have you tested it pretty thoroughly?  (I'm just concerned that somehow the order or operations mattered here.)

Perhaps what's confusing me is that `delete_id` seems to be a really confusing var name. It's actually the id of the new row to select, correct?

Reply:
The order won't matter as the only change now from the previous code is that the focused element changes before the draft gets deleted. Though I could move `remove_draft` above (won't make much difference but suits the order in which things seem to take place, i.e., the draft gets deleted and then the new draft is focused.

Reply:
Ok, seems reasonable!